src/circletexttest.c

```diff
@@ -26,12 +26,24 @@ int main(void)
 	in = fopen("eleanor.jpg", "rb");
 	if(!in) {
 		im = gdImageCreateTrueColor(300, 300);
+		if(!im) {
+			fprintf(stderr, "gdImageCreateTrueColor failed \n");
+			return 1;
+		}
 	} else {
 		im = gdImageCreateFromJpeg(in);
 		fclose(in);
+		if(!im) {
+			fprintf(stderr, "gdImageCreateFromJpeg failed \n");
+			return 1;
+		}
 	}
 #else
 	im = gdImageCreateTrueColor(300, 300);
+	if(!im) {
+		fprintf(stderr, "gdImageCreateTrueColor failed \n");
+		return 1;
+	}
 #endif /* HAVE_LIBJPEG */
```

Comment:
please put a single null pointer check here rather than copying & pasting it 3 times